content/200-orm/200-prisma-client/150-using-raw-sql/100-typedsql.mdx

```diff
@@ -0,0 +1,204 @@
+---
+title: 'TypedSQL'
+metaTitle: 'Writing Type-safe SQL with TypedSQL and Prisma Client'
+metaDescription: 'Learn how to use TypedSQL to write fully type-safe SQL queries that are compatible with any SQL console and Prisma Client.'
+sidebar_class_name: preview-badge
+---
+
+## Getting started with TypedSQL
+
+To start using TypedSQL in your Prisma project, follow these steps:
+
+1. Ensure you have `@prisma/client` and `prisma` installed and updated to at least version `5.19.0`.
+
+1. Add the `typedSql` preview feature flag to your `schema.prisma` file:
+
+   ```prisma
+   generator client {
+     provider = "prisma-client-js"
+     previewFeatures = ["typedSql"]
+   }
+   ```
+
+1. Create a `sql` directory inside your `prisma` directory. This is where you'll write your SQL queries.
+
+   ```terminal
+   cd prisma
+   mkdir sql
+   ```
+
+1. Create a new `.sql` file in your `prisma/sql` directory. For example, `getUsersWithPosts.sql`. Note that the file name must be a valid JS identifier and cannot start with a `$`.
```

Comment:
Maybe link to what a "JS indentifier" is in w.e docs that would be? (probably overkill)